daemon/src/daemon.rs

```diff
@@ -872,13 +883,45 @@ impl Daemon {
     }
 }
 
+// Spawn a file watcher and feed its events to the document_handle.
+// In addition, a short timeout after the last event, do a full re-scan, so that we don't miss any
+// file changes - the watcher isn't necessarily exhaustive.
 async fn spawn_file_watcher(base_dir: &Path, document_handle: DocumentActorHandle) {
     let mut watcher = Watcher::new(base_dir);
+
     tokio::spawn(async move {
-        while let Some(watcher_event) = watcher.next().await {
-            document_handle
-                .send_message(DocMessage::FromWatcher(watcher_event))
-                .await;
+        let debounce_duration = Duration::from_millis(100);
+
+        let debounce_timer = tokio::time::sleep(debounce_duration);
+        // Sleep does not implement the Unpin trait, so in order to use it with select!, we have to
+        // pin it first (according to the documentation).
```

Comment:
```suggestion
        // pin it first (according to the documentation https://docs.rs/tokio/latest/tokio/time/struct.Sleep.html).
```
What do you think about linking the documentation? That would make it for me smoother to read if I get back to this later... Downside is, the link might become outdated.